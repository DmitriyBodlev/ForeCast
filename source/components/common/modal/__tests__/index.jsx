import React from 'react';
import { shallow } from 'enzyme';
import { ModalComponent, HeaderComponent, FooterComponent } from '../';

describe('source/components/common/modal', () => {
  describe('modal component', () => {
    it('should render component', () => {
      const closeModal = jest.fn();
      const modalParams = {
        modal: {
          component: <div>some text</div>,
          options: {
            position: 'top-right',
            title: 'text',
            controlButtons: [],
          },
        },
      };
      const wrapper = shallow(
        <ModalComponent
          closeModal={closeModal}
          params={modalParams} />,
      );
      const modalBody = wrapper.contains(<div>some text</div>);

      expect(wrapper).toMatchSnapshot();
      expect(modalBody).toBeTruthy();
    });

    it('should render component with control button', () => {
      const closeModal = jest.fn();
      const buttonAction = jest.fn();
      const modalParams = {
        modal: {
          component: <div>some text</div>,
          options: {
            position: 'top-left',
            width: 600,
            height: 400,
            controlButtons: [
              { name: 'Save', type: 'primary', action: buttonAction },
            ],
          },
        },
      };

      const wrapper = shallow(
        <ModalComponent
          closeModal={closeModal}
          params={modalParams} />,
      );

      const headerComponent = wrapper.find(HeaderComponent);
      const footerComponent = wrapper.find(HeaderComponent);

      expect(wrapper).toMatchSnapshot();
      expect(headerComponent).toBeTruthy();
      expect(footerComponent).toBeTruthy();
    });
  });

  describe('header component', () => {
    it('should render header component', () => {
      const closeModal = jest.fn();

      const wrapper = shallow(
        <HeaderComponent
          closeAction={closeModal}
          title='Test' />,
      );

      wrapper.find('a').simulate('click');

      expect(wrapper).toMatchSnapshot();
      expect(closeModal).toHaveBeenCalled();
    });
  });

  describe('footer component', () => {
    it('should render footer component', () => {
      const closeModal = jest.fn();
      const buttonAction = jest.fn();
      const buttonList = [
        { name: 'Save', type: 'primary', action: buttonAction },
      ];
      const wrapper = shallow(
        <FooterComponent
          closeAction={closeModal}
          buttons={buttonList} />,
      );

      const buttons = wrapper.find('button');
      const modalButtons = buttonList.length + 1;

      buttons.forEach((button: Object) => button.simulate('click'));

      expect(wrapper).toMatchSnapshot();
      expect(closeModal).toHaveBeenCalled();
      expect(buttonAction).toHaveBeenCalled();
      expect(buttons.length).toEqual(modalButtons);
    });
  });
});
