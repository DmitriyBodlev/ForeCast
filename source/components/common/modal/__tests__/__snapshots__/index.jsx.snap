// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source/components/common/modal footer component should render footer component 1`] = `
<styled(Component)>
  <button
    className="primary"
    key="0"
    onClick={[Function]}
  >
    Save
  </button>
  <button
    onClick={[Function]}
  >
    Cansel
  </button>
</styled(Component)>
`;

exports[`source/components/common/modal header component should render header component 1`] = `
<WithTheme(styled(Component))>
  <styled(Component)>
    Test
  </styled(Component)>
  <a
    onClick={[Function]}
  >
    <Connect(LabelComponent)
      endLabel="shape"
    />
  </a>
</WithTheme(styled(Component))>
`;

exports[`source/components/common/modal modal component should render component 1`] = `
<styled(Component)>
  <WithTheme(styled(Component))
    className="css-m5r4ys"
    params={
      Object {
        "controlButtons": Array [],
        "position": "top-right",
        "title": "text",
      }
    }
  >
    <HeaderComponent
      closeAction={[Function]}
      title="text"
    />
    <div>
      some text
    </div>
  </WithTheme(styled(Component))>
</styled(Component)>
`;

exports[`source/components/common/modal modal component should render component with control button 1`] = `
<styled(Component)>
  <WithTheme(styled(Component))
    className="css-1qhj9c7"
    params={
      Object {
        "controlButtons": Array [
          Object {
            "action": [Function],
            "name": "Save",
            "type": "primary",
          },
        ],
        "height": 400,
        "position": "top-left",
        "width": 600,
      }
    }
  >
    <HeaderComponent
      closeAction={[Function]}
    />
    <div>
      some text
    </div>
    <FooterComponent
      buttons={
        Array [
          Object {
            "action": [Function],
            "name": "Save",
            "type": "primary",
          },
        ]
      }
      closeAction={[Function]}
    />
  </WithTheme(styled(Component))>
</styled(Component)>
`;
